### ISSUE TITLE
Do you think this fix could be integrated in Sarissa? I have a Sarissa branch in GitHub.

### ISSUE THREAD

**Issue by volkertb** [mareksuscak/jquery-sarissa-compat#1]
Hi Marek!

First of all, splendid work! I've been hassling with Sarissa/jQuery/AJAX4JSF myself as well, so this work is much appreciated.

The apparent lack of recent activity in the upstream Sarissa project on SourceForge (plus the fact that the code base is still maintained in a CVS repository) led me export the CVS repository to Git and start a branch on GitHub myself. I don't intend to make it an official fork at this time, but I would like to see the project invigorated, since many people (including you and me) apparently still rely on Sarissa in legacy web applications and need it to work with newer web browsers.

I was thinking: could your patch/fix somehow be integratd in Sarissa, so that it would become compatible with jQuery out-of-the-box? Since jQuery is also a very popular JavaScript library, it would make a lot of sense if Sarissa were to be compatible with it without any add-ons.

I understand that it would require some refactoring of your fix, since your current code is meant to b loaded _after_ jQuery, whereas Sarissa is often loaded _before_ jQuery, such as in the case of RichFaces and AJAX4JSF.

What do you think? If you would like to give it a shot, please have a look at the Sarissa 0.9.9.7-SNAPSHOT branch that I started on GitHub: https://github.com/volkertb/sarissa/tree/0.9.9.7-SNAPSHOT

So far, I've committed some fixes to make Maven build the project again, got rid of build warnings, made the Site plugin (including the JavaScript/ECMAScript unit test runner) work again, etc.

I intend to create a Gist of the resulting sarissa.js with version number 0.9.9.7-SNAPSHOT, since the latest CVS trunk version of sarissa.js apparently contained some fixes since 0.9.9.6, even though the version number was never raised since.

Please let me know what you think. If you think my proposal is possible and if I could help you with integrating your jQuery compatibility fix in Sarissa, feel free to let me know.

Thanks for considering this and keep on coding! :-)

**Comment by mareksuscak** [comment]
Hey @volkertb. First and foremost, thank you for the kind words. Secondly, I created this project about two years ago while developing custom extensions for Salesforce which is seemingly built on the JSF and thus makes use of AJAX4JSF and the accompanying Sarissa library. It was a one-off project and I'm not working on it anymore hence with all due respect it is not worth to invest into further development at moment. Additionally it might not be possible at all to patch this snag the way you requested (when loading Sarissa before jQuery) for a number of (not only) compatibility reasons. Besides that I made some effort and took a look at Sarissa's current code - turns out it no longer exposes `originalXmlHttpRequest` - and I am not sure why. The implementation once exposed it - http://www.rbecker.eu/2014/08/cors-xmlhttprequest-in-ie10-and-newer.html.

**Comment by volkertb** [comment]
Yeah, I noticed that as well. Apparently, the version of sarissa.js that was bundled with RichFaces/AJAX2jsf stored the original reference to the native XmlHttpRequest object in Sarissa.originalXmlHttpRequest, but the upstream code did not. It appears that the JBoss/RichFaces developers either never bothered to offer that feature upstream, or it was never accepted by the Sarissa maintainers. Anyway, I added that feature back into my 0.9.9.7-SNAPSHOT branch yesterday, so the version in my branch is now compatible again with your library. :+1:

You're probably right about your patch not being suitable for integration in Sarissa, since it indeed requires jQuery to be loaded before Sarissa, which is often not the case, especially when it is embedded in other frameworks and software such as RichFaces and Salesforce.

I know exactly what you mean with having to start these projects to work around issues at work. I'm currently working at a customer who's application I'm maintaining is still dependent on RichFaces 3.x, hence my problems with Sarissa. But I think that the version I have in my branch works pretty well for all recent versions of Internet Explorer now. And just to be sure, I added your library to my customer's application as well. I appreciate your work on that! :+1: 

Thanks for your feedback and have fun with whatever you do!
